src/reconcile/rec_write.c

```diff
@@ -2554,6 +2554,12 @@ __rec_write_wrapup(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_PAGE *page)
         break;
     }
 
+    /* If the page has post-instantiation delete information, we don't need it any more. */
+    if (mod->instantiated) {
```

Comment:
Is it safe to free? We may in checkpoint.

Reply:
I think so; unless there's a way the same page can be getting reconciled twice at the same time, and I thought that wasn't supposed to happen. AFAIK, the checkpoint tree-walk holds a hazard pointer that will prevent eviction from visiting the same page simultaneously, and eviction itself locks the target page.